torch/csrc/jit/codegen/cuda/lower_index_compute.cpp

```diff
@@ -268,6 +268,179 @@ IndexingParameters getNonGlobalInitialIndexParameters(
   return index_parameters;
 }
 
+//! Initial index parameters for predicate, adjusts loop to indexing
+//!  may according to the information annotated on the loop nest.
+//!
+//! TODO:
+//!  This function is mostly copy pasted from previous implementation
+//! at this step, further clean up is possible since:
+//!  1. Much of the loop-to-ind adjustment will be issued from idgraph
+//!  2. Much of the initial index logic could be shared across all
+//! the 3 variants.
+IndexingParameters getPredicateInitialIndexParameters(
```

Comment:
Copy and paste mostly for this function body. Further detailed cleanup will be on follow ups, to keep this PR small.